.github/workflows/CI-tag-orchestrator.yml

```diff
@@ -5,54 +5,132 @@ on:
     tags:
       - "[0-9]+.[0-9]+.[0-9]+-[0-9]+.[0-9]+.[0-9]+*"
   workflow_dispatch:
+    inputs:
+      git_tag:
+        type: string
+    secrets:
+      DOCKER_HUB_USERNAME:
+        required: true
+      DOCKER_HUB_TOKEN:
+        required: true
 
 env:
   RUST_BACKTRACE: 1
 
 jobs:
-  execute-system-tests:
-    name: Execute System Tests
-    uses: HorizenLabs/zkVerify-qa/.github/workflows/CI-e2e-tests.yml@main
+  env-setup:
+    uses: ./.github/workflows/CI-build.yml
     with:
-      zkverify_version: ${{ github.ref_name }}
-    secrets:
-      DOCKER_HUB_USERNAME: ${{ secrets.DOCKER_HUB_USERNAME }}
-      DOCKER_HUB_TOKEN: ${{ secrets.DOCKER_HUB_TOKEN }}
-      QA_SLACK_WEBHOOK_URL: ${{ secrets.QA_SLACK_WEBHOOK_URL }}
-      GH_TOKEN: ${{ secrets.GH_TOKEN }}
-
-  build-and-publish:
-    name: Build and Publish Docker Image
+      CACHING_MODE: "CACHE_RESTORE"
+  
+  build:
     runs-on: ubuntu-latest
-    needs: execute-system-tests # Wait for e2e tests to finish
+    needs: env-setup
+
+    outputs:
+      artifact_name_solo: ${{ steps.create-tar-file.outputs.tar_file_name_solo }}
+      artifact_name_relay: ${{ steps.create-tar-file.outputs.tar_file_name_relay }}
```

Comment:
May not be needed if using ENV vars for those values since they are static :shrug: Look at the comments below

Reply:
They need to be outputs to be able to be used in a different job 
https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#passing-values-between-steps-and-jobs-in-a-workflow

Reply:
I see why now. Your gitt_tag was in fact "" empty but build-and-test job is defaulting to github_ref..... tag coming from the job if input is empty, smth to keep in mind if this will not step on the way. When you manually INPUT one tag but `build-and-test` job will use a diff tag to build the project

![image](https://github.com/user-attachments/assets/1a18decc-e491-40ad-bb4d-461263074a75)

```
steps:
      - uses: actions/checkout@v4
        with:
          ref: ${{ github.event.inputs.git_tag || github.ref }}
          fetch-depth: 0
```